### DISCUSSION TITLE
tsh login doesn't work with mac touch ID

### DISCUSSION THREAD

**Discussion by hydezhao** [gravitational/teleport#18206] [Q&A] [answered]
I've setup my account with touch ID as MFA via webUI. While when I try to do `tsh login` in terminal, it seems terminal is not reading/waiting for input from touch ID. 

Note that if the mfa is setup with a Yubikey, it works well with `tsh login`

**Comment by zmb3** [comment, marked as answer]
For TouchID, registration using the web UI and registration using `tsh` are treated as separate devices. If you want to use Touch ID with `tsh`, you'll just need to register it using `tsh mfa add`.

Did I get that right, @codingllama?

**Reply by hydezhao** [reply]
When I try `tsh mfa add` the terminal is not prompting to accept touch id input.

**Reply by zmb3** [reply]
How did you install `tsh`? In order to use Touch ID you need to download the signed and notarized builds from goteleport.com/download. Building from source or downloading via something like Homebrew won't support Touch ID.

**Reply by hydezhao** [reply]
ehh I installed via homebrew.. I will try download from official page.

**Reply by hydezhao** [reply]
thx a lot @zmb3 , by using the official package it works !

**Reply by zmb3** [reply]
Awesome, glad to hear it!

**Reply by codingllama** [reply]
Glad to see it all worked out!

The need for the repeat Web/tsh registration is a technical gap we can't (yet) cross, sadly. As for Homebrew, the open source build cannot use our Apple Developer key (for obvious reasons), so that makes it impossible for it do to Touch ID.

Leaving a few links here for future reference:

* Web/tsh registration explanation: https://goteleport.com/docs/access-controls/guides/passwordless/#step-12-register
* Touch ID not usable: https://goteleport.com/docs/access-controls/guides/passwordless/#touch-id-not-usable
